### PR TITLE
Fix various small issues and spelling errors

### PR DESCRIPTION
**Pull request by sandoba**
A few small fixes for language, formatting and code/references.

### CODE REVIEW

File: Documentation/Configuration/Typo3ConfVars/SYS.rst [hunk 1]
```diff
@@ -950,7 +950,7 @@ linkHandler
     *   `url` (see :t3src:`core/Classes/LinkHandling/UrlLinkHandler.php`)
     *   `email` (see :t3src:`core/Classes/LinkHandling/EmailLinkHandler.php`)
     *   `record` (see :t3src:`core/Classes/LinkHandling/RecordLinkHandler.php`)
-    *   `phone` (see :t3src:`core/Classes/LinkHandling/TelephoneLinkHandler.php`)
```
Comment (linawolf) [inline Documentation/Configuration/Typo3ConfVars/SYS.rst:953]: I think these are not words but fixed strings to be used for configuration. Do you have a reference for this change?

File: Documentation/Configuration/Typo3ConfVars/SYS.rst [hunk 2]
```diff
@@ -610,7 +610,7 @@ productionExceptionHandler
    Note: The configured "productionExceptionHandler" is used if
    :ref:`[SYS][displayErrors]<typo3ConfVars_sys_displayErrors>` is set to "0"
    or is set to "-1" and
-   :ref:`[SYS][devIPmask]<typo3ConfVars_sys_devIPmask>` doesnt match the user's IP.
+   :ref:`[SYS][devIPmask]<typo3ConfVars_sys_devIPmask>` doesn't match the user's IP.
```
Comment (linawolf) [inline Documentation/Configuration/Typo3ConfVars/SYS.rst:613]:
```suggestion
   :ref:`[SYS][devIPmask]<typo3ConfVars_sys_devIPmask>` does not match the user's IP.
```
"We are striving to utilize non-colloquial English, abstaining from unnecessary shortening of words."

This maintains clarity and formality, which can be beneficial when writing documentation for an international audience.

File: Documentation/Configuration/Typo3ConfVars/MAIL.rst [hunk 3]
```diff
@@ -250,8 +251,8 @@ transport_smtp_stream_options
 .. confval:: $GLOBALS['TYPO3_CONF_VARS']['MAIL']['transport_smtp_stream_options']
 
    :Path: $GLOBALS['TYPO3_CONF_VARS']['MAIL']
-   :type: bool
```
Comment (linawolf) [inline Documentation/Configuration/Typo3ConfVars/MAIL.rst:253]: I think both bool and array are allowed...?

File: Documentation/Configuration/Typo3ConfVars/MAIL.rst [hunk 4]
```diff
@@ -167,7 +168,7 @@ transport
       setting.
 
    mbox
-      This doesnt send any mail out, but instead will write every outgoing mail
+      This doesn't send any mail out, but instead will write every outgoing mail
```
Comment (linawolf) [inline Documentation/Configuration/Typo3ConfVars/MAIL.rst:171]:
```suggestion
      This does not send any mail out, but instead will write every outgoing mail
```

**Review by linawolf — COMMENTED** [review]
Thanks for your changes. Just some small remarks (see comments and suggestions in code)

For the next time:  is a good idea to keep different topics separate, so the can be discussed and committed separately.

File: Documentation/Configuration/Typo3ConfVars/MAIL.rst (hunk 3 above)
Comment (sandoba) [reply Documentation/Configuration/Typo3ConfVars/MAIL.rst:253]: The only use of this setting in code I can see is this: $streamOptions = (array)($mailSettings['transport_smtp_stream_options'] ?? []);

"DefaultConfigurationDescription.yaml" uses "mixed" as type, "DefaultConfiguration.php" uses "null" as value, the unit test uses "array", the documentation here uses "bool" as type but the example is "array". So perhaps "mixed" would be ok but no other setting for the configuration in the documentation has "mixed" listed at the moment, but there are settings with "array"+"null".

File: Documentation/Configuration/Typo3ConfVars/SYS.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/Configuration/Typo3ConfVars/SYS.rst:953]: Ack

File: Documentation/Configuration/Typo3ConfVars/MAIL.rst (hunk 3 above)
Comment (linawolf) [reply Documentation/Configuration/Typo3ConfVars/MAIL.rst:253]: Ack

**Review by linawolf — APPROVED** [review]

**Comment by linawolf** [comment]
Thanks for your suggestions

**Pull request merged**